### PR TITLE
Add GLPI detection

### PR DESCRIPTION
**Pull request by Darkiros**
Add GLPI detection.
If version is `>=10`, version is set to 10 because since 10.0 it is hard to know exactly the version.

Here is the github project of GLPI : https://github.com/glpi-project/glpi

### CODE REVIEW

File: src/technologies/g.json [hunk 1]
```diff
@@ -25,6 +25,23 @@
...
+    "cats": [
+        13
+    ],
+    "cpe": "cpe:2.3:a:glpi-project:glpi:*:*:*:*:*:*:*:* ",
+    "description": "GLPI is a Free Asset and IT Management Software package, Data center management, ITIL Service Desk, licenses tracking and software auditing.",
+    "html": [
+        "<link [^>]*href=[^>]*v=((\\d\\.?)*)[^a-z]\\;version:\\1?\\1:10",
+        "<script [^>]*src=[^>]*v=((\\d\\.?)*)[^a-z]\\;version:\\1?\\1:10",
+        "<a[^>]*\\DPowered By [^>]*\\D>GLPI version ((\\d\\.?)*)\\;version:\\1",
+        "<input[^>]*name=\\D_glpi_csrf_token\\D"
```
Comment (nurbek91) [inline src/technologies/g.json:38]:
```
 "dom": {
      "input[name='_glpi_csrf_token']": {
        "attributes": {
          "name": ""
        }
      },
      "div[id*='footer-login'] > a": {
          "text": "GLPI\\sversion\\s+([\\d\\.]+)\\;version:\\1"
      }
    },
```

File: src/technologies/g.json [hunk 2]
```diff
@@ -25,6 +25,23 @@
     "saas": true,
     "website": "https://www.geovendas.com"
   },
+  "GLPI": {
+    "cats": [
+        13
```
Comment (nurbek91) [inline src/technologies/g.json:30]: Maybe add CRM category but i am not sure

File: src/technologies/g.json (hunk 1 above)
Comment (nurbek91) [reply src/technologies/g.json:38]: Maybe we can add pricing also 
https://glpi-project.org/prices/
```
"pricing": [
      "mid",
      "recurring"
    ],
```

**Review by nurbek91 — CHANGES_REQUESTED** [review]
Hey mate, thanks for PR!! Need some updates to bring ideal and resource-intensive.
Examples:
http://mquest.com.br/
http://veapp.net/

Are there any examples where versions above 10 are used?

**Comment by Darkiros** [comment]
> Hey mate, thanks for PR!! Need some updates to bring ideal and resource-intensive. Examples: http://mquest.com.br/ http://veapp.net/
> 
> Are there any examples where versions above 10 are used?

Hi, here you can find GLPI versions above 10:
https://glpi.webcom-info.com/

File: src/technologies/g.json (hunk 1 above)
Comment (Darkiros) [reply src/technologies/g.json:38]:
```
"dom": {
      "input[name='_glpi_csrf_token']": {
        "attributes": {
          "name": ""
        }
      },
      "div[id*='footer-login'] > a": {
          "text": "GLPI\\sversion\\s+([\\d\\.]+)\\;version:\\1"
      }
    },
```
I don't know why but with `dom` section it doesn't seem to work.

File: src/technologies/g.json (hunk 2 above)
Comment (Darkiros) [reply src/technologies/g.json:30]: Yes, I think it might fits too.

File: src/technologies/g.json (hunk 1 above)
Comment (nurbek91) [reply src/technologies/g.json:38]: Hm strange. I tested on versions below 10 and everything worked there (the part with checking versions and just checking the input). Now I will look at your links above 10 versions

**Comment by nurbek91** [comment]
> > Hey mate, thanks for PR!! Need some updates to bring ideal and resource-intensive. Examples: http://mquest.com.br/ http://veapp.net/
> > Are there any examples where versions above 10 are used?
> 
> Hi, here you can find GLPI versions above 10: https://glpi.webcom-info.com/ https://glpi.picinformatique.com/ https://glpi.sqli.com/

Above 10 versions are defined without regular expressions by these traces

```
"js": {
      "glpiUnsavedFormChanges": ""
    },
"meta": {
      "glpi:csrf_token": ""
    },
```

File: src/technologies/g.json (hunk 1 above)
Comment (Darkiros) [reply src/technologies/g.json:38]:
```
 "div[id*='footer-login'] > a": {
          "text": "GLPI\\sversion\\s+([\\d\\.]+)\\;version:\\1"
      }
```
This one doesn't seem work on http://mquest.com.br/ for me.

Comment (nurbek91) [reply src/technologies/g.json:38]: > ```
>  "div[id*='footer-login'] > a": {
>           "text": "GLPI\\sversion\\s+([\\d\\.]+)\\;version:\\1"
>       }
> ```
> 
> This one doesn't seem work on http://mquest.com.br/ for me.

For now, you can make other changes. I'll test again in that case tomorrow mate

**Review by nurbek91 — CHANGES_REQUESTED** [review]
Tested, everything works.
Can you make it look like this?  need to remove the html and change the regex a bit
```
"GLPI": {
    "cats": [
        18,
        53
    ],
    "cpe": "cpe:2.3:a:glpi-project:glpi:*:*:*:*:*:*:*:* ",
    "description": "GLPI is an open-source IT Asset Management, issue tracking and service desk system.",
    "dom": {
      "input[name='_glpi_csrf_token']": {
        "attributes": {
          "name": ""
        }
      },
        "div[id='footer-login'] > a": {
            "text": "GLPI\\s+version\\s+([\\d\\.]+)\\;version:\\1"
        }
    },
    "js": {
        "glpiUnsavedFormChanges": ""
    },
    "meta": {
        "glpi:csrf_token": ""
    },
    "oss": true,
    "icon": "GLPI.png",
    "implies": "PHP",
    "pricing": [
        "mid",
        "recurring"
      ],
    "website": "https://glpi-project.org"   
  },
```

**Comment by Darkiros** [comment]
> Tested, everything works. Can you make it look like this? need to remove the html and change the regex a bit
> 
> ```
> "GLPI": {
>     "cats": [
>         18,
>         53
>     ],
>     "cpe": "cpe:2.3:a:glpi-project:glpi:*:*:*:*:*:*:*:* ",
>     "description": "GLPI is an open-source IT Asset Management, issue tracking and service desk system.",
>     "dom": {
>       "input[name='_glpi_csrf_token']": {
>         "attributes": {
>           "name": ""
>         }
>       },
>         "div[id='footer-login'] > a": {
>             "text": "GLPI\\s+version\\s+([\\d\\.]+)\\;version:\\1"
>         }
>     },
>     "js": {
>         "glpiUnsavedFormChanges": ""
>     },
>     "meta": {
>         "glpi:csrf_token": ""
>     },
>     "oss": true,
>     "icon": "GLPI.png",
>     "implies": "PHP",
>     "pricing": [
>         "mid",
>         "recurring"
>       ],
>     "website": "https://glpi-project.org"   
>   },
> ```

It works for the dom, I missed something sorry.
But if we remove the html, we will only be able to detect the version 0.X.X of GLPI not 9.X.X for exemple.

**Comment by nurbek91** [comment]
> 9.X.X for exemple
Ah ok. can u share links i will take a look

**Comment by Darkiros** [comment]
> > 9.X.X for exemple
> 
> Ah ok. can u share links i will take a look

Sure, this one for exemple:
https://glpi.vigians.com/

**Comment by nurbek91** [comment]
> > > 9.X.X for exemple
> > 
> > 
> > Ah ok. can u share links i will take a look
> 
> Sure, this one for exemple: https://glpi.vigians.com/ https://glpi.actuafilms.com/
We can add something like this, 
`"scriptSrc": "//glpi\\..+common\\.min\\.js\?v=([\\d\\.]+)\\;version:\\1",`
the main thing is to remove html and it is desirable not to use dom either.
Another question is can we specify requires: "Debian" ?

**Comment by Darkiros** [comment]
> > > > 9.X.X for exemple
> > > 
> > > 
> > > Ah ok. can u share links i will take a look
> > 
> > 
> > Sure, this one for exemple: https://glpi.vigians.com/ https://glpi.actuafilms.com/
> 
> We can add something like this, `"scriptSrc": "//glpi\\..+common\\.min\\.js\?v=([\\d\\.]+)\\;version:\\1",` the main thing is to remove html and it is desirable not to use dom either. Another question is can we specify requires: "Debian" ?

Yes it works but it is quite restricted because this will only work on glpi.XX.XX domain names.

About debian, we can't specify it because GLPI can be installed everywhere, you just need PHP.

**Review by nurbek91 — APPROVED** [review]
Hey mate , thanks!

**Comment by Darkiros** [comment]
> Hey mate , thanks!

Hi ! This is the final commit, I changed the regex to : 
`"scriptSrc": "//.*glpi.+common\\.min\\.js\\?v=(\\d+\\.\\d+\\.\\d+)\\;version:\\1",`
Because the last one matched with version >=10 and wrote a false number in version. Now it should be fixed.

**Review by nurbek91 — APPROVED** [review]

**Comment by AliasIO** [comment]
Thanks!

**Pull request merged**